pkg/controller/vpc_nat_gw_nat.go

```diff
@@ -1658,3 +1703,42 @@ func (c *Controller) isDnatDuplicated(eipName, dnatName, externalPort string) (b
 	}
 	return false, nil
 }
+
+func (c *Controller) createOrUpdateCrdFip(key, eipName, internalIp string) error {
+	eipCr, err := c.config.KubeOvnClient.KubeovnV1().IptablesFIPRules().Get(context.Background(), key, metav1.GetOptions{})
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			_, err := c.config.KubeOvnClient.KubeovnV1().IptablesFIPRules().Create(context.Background(), &kubeovnv1.IptablesFIPRule{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: key,
+				},
+				Spec: kubeovnv1.IptablesFIPRuleSpec{
+					EIP:        eipName,
+					InternalIp: internalIp,
+				},
+			}, metav1.CreateOptions{})
+
+			if err != nil {
+				errMsg := fmt.Errorf("failed to create fip crd %s, %v", key, err)
+				klog.Error(errMsg)
+				return errMsg
+			}
+		} else {
+			errMsg := fmt.Errorf("failed to get fip crd %s, %v", key, err)
+			klog.Error(errMsg)
+			return errMsg
+		}
+	} else {
+		eip := eipCr.DeepCopy()
```

Comment:
no need for deepcopy and can move it into the if block

Reply:
fixed